### PR TITLE
add requirement.md

### PR DESCRIPTION
**Pull request by ramsayleung**
1. add requirement for admin-web.

### CODE REVIEW

File: docs/requirement.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## 2. Requirement
+
+### 2.1. basic requirement about productType
+1. name and description of productType are required, others is optional
+2. type, label, required, attributeConstraint, searchable in attributes of
+productType are required
+3. name in attributes of category should be alphabetic, numeric, underscore(_) and hyphen(-)
+characters.Maximize size of slug is 256, and minimum size is 2.
+
```
Comment (davis999) [inline docs/requirement.md]: 1. multi language.
2. multi attribute.
3. multi attribute constraint.

File: docs/requirement.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+### 2.2. Create productType
+1. name in attributes of productType should be unique
+
+### 2.3. Delete productType
+1. version and id of productType should match
```
Comment (davis999) [inline docs/requirement.md:28]: can not delete producttype when there is product associate with it.

File: docs/requirement.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+### 2.4. Update productType
+1. all attributes of productType could be updated
+2. version and id of productType should match
+
```
Comment (davis999) [inline docs/requirement.md]: 1. if remove attribute, associated product should remove the attribute value.
2. if add attribute, associated product should add the attribute value.
3. con not change attribute constraint.

File: docs/requirement.md (hunk 1 above)
Comment (ramsayleung) [reply docs/requirement.md]: 1. add multi-language demonstration 
2. add multi-attribute demonstration
3. add constrain demonstration for each attribute

File: docs/requirement.md (hunk 2 above)
Comment (ramsayleung) [reply docs/requirement.md:28]: 1. add demonstration

File: docs/requirement.md (hunk 3 above)
Comment (ramsayleung) [reply docs/requirement.md]: 1. add demonstration about these

**Review by davis999 — APPROVED** [review]

**Pull request merged**